yamux/src/session.rs

```diff
@@ -618,50 +618,55 @@ where
             );
         }
 
-        while let Some(ref mut interval) = self.keepalive {
+        if let Some(ref mut interval) = self.keepalive {
             match Pin::new(interval).as_mut().poll_next(cx) {
                 Poll::Ready(Some(_)) => {
                     self.keep_alive(cx, Instant::now())?;
                 }
                 Poll::Ready(None) => {
                     debug!("yamux::Session poll keepalive interval finished");
-                    break;
                 }
-                Poll::Pending => break,
+                Poll::Pending => (),
             }
         }
 
-        loop {
+        let mut need_wake = false;
+        #[allow(unused_assignments)]
```

Comment:
unnecessary lint?

Reply:
init value unused, cargo will report an warn

Reply:
doesn't used in line 664?
```
if need_wake {
```

Reply:
633 init, 642 resets, cargo think this is not good

Reply:
what's your rust version? I can't reproduce this warning when run with `make clippy`

Reply:
...I can’t reproduce it either. It may be that an error was reported during the modification process, but the final version is no longer needed. I will delete it.